docs/tutorials/go-built-in.md

```diff
@@ -103,579 +101,527 @@ func NewKVStoreApplication() *KVStoreApplication {
+	"syscall"
+
+	"github.com/dgraph-io/badger"
+	"github.com/spf13/viper"
+	abciclient "github.com/tendermint/tendermint/abci/client"
+	cfg "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/libs/log"
+	nm "github.com/tendermint/tendermint/node"
+	"github.com/tendermint/tendermint/types"
 )
 
-var configFile string
+var homeDir string
 
 func init() {
- flag.StringVar(&configFile, "config", "$HOME/.tendermint/config/config.toml", "Path to config.toml")
+	flag.StringVar(&homeDir, "tm-home", "", "Path to the tendermint 'home' directory")
 }
 
 func main() {
- db, err := badger.Open(badger.DefaultOptions("/tmp/badger"))
- if err != nil {
-  fmt.Fprintf(os.Stderr, "failed to open badger db: %v", err)
-  os.Exit(1)
- }
- defer db.Close()
- app := NewKVStoreApplication(db)
-
- flag.Parse()
-
- node, err := newTendermint(app, configFile)
- if err != nil {
-  fmt.Fprintf(os.Stderr, "%v", err)
-  os.Exit(2)
- }
-
- node.Start()
- defer func() {
-  node.Stop()
-  node.Wait()
- }()
-
- c := make(chan os.Signal, 1)
- signal.Notify(c, os.Interrupt, syscall.SIGTERM)
- <-c
-}
-
-func newTendermint(app abci.Application, configFile string) (*nm.Node, error) {
- // read config
- config := cfg.DefaultValidatorConfig()
- config.RootDir = filepath.Dir(filepath.Dir(configFile))
- viper.SetConfigFile(configFile)
- if err := viper.ReadInConfig(); err != nil {
-  return nil, fmt.Errorf("viper failed to read config file: %w", err)
- }
- if err := viper.Unmarshal(config); err != nil {
-  return nil, fmt.Errorf("viper failed to unmarshal config: %w", err)
- }
- if err := config.ValidateBasic(); err != nil {
-  return nil, fmt.Errorf("config is invalid: %w", err)
- }
-
- // create logger
- logger := log.NewTMLogger(log.NewSyncWriter(os.Stdout))
- var err error
- logger, err = tmflags.ParseLogLevel(config.LogLevel, logger, cfg.DefaultLogLevel)
- if err != nil {
-  return nil, fmt.Errorf("failed to parse log level: %w", err)
- }
-
- // read private validator
- pv := privval.LoadFilePV(
-  config.PrivValidatorKeyFile(),
-  config.PrivValidatorStateFile(),
- )
-
- // read node key
- nodeKey, err := p2p.LoadNodeKey(config.NodeKeyFile())
- if err != nil {
-  return nil, fmt.Errorf("failed to load node's key: %w", err)
- }
-
- // create node
- node, err := nm.NewNode(
-  config,
-  pv,
-  nodeKey,
-  abcicli.NewLocalClientCreator(app),
-  nm.DefaultGenesisDocProviderFunc(config),
-  nm.DefaultDBProvider,
-  nm.DefaultMetricsProvider(config.Instrumentation),
-  logger)
- if err != nil {
-  return nil, fmt.Errorf("failed to create new Tendermint node: %w", err)
- }
-
- return node, nil
+	flag.Parse()
+	if homeDir == "" {
+		h, err := os.UserHomeDir()
+		if err != nil {
+			panic(err)
+		}
+		homeDir = fmt.Sprintf("%s/%s", h, ".tendermint")
+	}
+	config := cfg.DefaultValidatorConfig()
+
+	config.SetRoot(homeDir)
+
+	viper.SetConfigFile(fmt.Sprintf("%s/%s", homeDir, "config/config.toml"))
+	if err := viper.ReadInConfig(); err != nil {
+		panic(err)
```

Comment:
These panics should probably all be `log.Fatal` instead; a goroutine dump isn't valuable in debugging "expected" infra failures.

```suggestion
		log.Fatalf("Reading config: %v", err)
```